src/components/timecard/edit-shift-timecard/EditShiftTimecard.jsx

```diff
@@ -2,19 +2,39 @@ import { PropTypes } from 'prop-types';
 import { useState } from 'react';
 import { Link } from 'react-router-dom';
 import EditShiftHours from '../edit-shift-hours/EditShiftHours';
+import { deleteTimeEntry } from '../../../api/services/timecardService';
+import { UrlSearchParamBuilder } from '../../../utils/api-utils/UrlSearchParamBuilder';
+import { useTimecardContext } from '../../../context/TimecardContext';
+import { deepClone } from '../../../utils/common-utils/common-utils';
 
 const EditShiftTimecard = ({ timeEntry, timeEntriesIndex }) => {
   const toggleEditShiftHours = (event) => {
     event.preventDefault();
     setShowEditShiftHours(!showEditShiftHours);
   };
 
-  const timeEntryIsEmpty =
+  const { timeEntries, setTimeEntries } = useTimecardContext();
+
+  const timeEntryExists =
     'startTime' in timeEntry && timeEntry.startTime !== '';
   const [showEditShiftHours, setShowEditShiftHours] = useState(
-    !timeEntryIsEmpty
+    !timeEntryExists
   );
 
+  const clickRemoveShiftButton = async (event) => {
```

Comment:
This function name sounds like it is automating the action of clicking the remove button.
Might be better named as something a bit more conventional e.g. 'handleClickRemoveShiftButton' or 'handleRemoveShift' etc.

Reply:
I've updated this